data/scripts/molgenis-model-update/update.py

```diff
@@ -0,0 +1,282 @@
+import shutil
+import os
+import pandas as pd
+import logging
+
+
+def float_to_int(df):
+    """
+    Cast float64 Series to Int64.
+    """
+    for column in df.columns:
+        if df[column].dtype == 'float64':
+            df.loc[:, column] = df[column].astype('Int64')
+
+    return df
+
+
+class TransformGeneral:
+    """General functions to update catalogue data model.
+    """
+
+    def __init__(self, database, database_type):
+        self.database = database
+        self.database_type = database_type
+        self.path = './' + self.database + '_data/'
+        self.logger = logging.getLogger(' data update and transform')
+
+    def delete_data_model_file(self):
+        """Delete molgenis.csv
+        """
+        os.remove(self.path + 'molgenis.csv')
+
+    def update_data_model_file(self):
+        """Replace data model file
+        """
+        # copy updated molgenis.csv to os.listdir(self.path)
+        if self.database_type == 'catalogue':
+            shutil.copy(os.path.abspath('../../datacatalogue3/molgenis.csv'),
+                        os.path.abspath(self.path))
+        elif self.database_type == 'network':
+            shutil.copy(os.path.abspath('../../datacatalogue3/stagingNetworks/molgenis.csv'),
+                        os.path.abspath(self.path))
+        elif self.database_type == 'cohort':
+            shutil.copy(os.path.abspath('../../datacatalogue3/stagingCohorts/molgenis.csv'),
+                        os.path.abspath(self.path))
+        elif self.database_type == 'cohortUMCG':
+            shutil.copy(os.path.abspath('../../datacatalogue3/stagingCohortsUMCG/molgenis.csv'),
+                        os.path.abspath(self.path))
+
+
+class TransformDataCatalogue:
+    """Functions to update catalogue data model from 2.8 to 3.0.
+    """
+
+    def __init__(self, database):
+        self.database = database
+        self.path = './' + self.database + '_data/'
+        self.logger = logging.getLogger(' data update and transform')
+
+    def transform_data(self):
+        """Make changes per table
+        """
+        self.contacts()
+        self.tables()
+        self.variables()
+        self.variable_values()
+        self.repeated_variables()
+        self.dataset_mappings()
+        self.variable_mappings()
+        self.datasources()
+
+    def contacts(self):
+        """Merge Contributions & Contacts on firstName and surname and rename columns
+        """
+        df_contributions = pd.read_csv(self.path + 'Contributions.csv')
+        df_contributions.rename(columns={'contributionType': 'role',
```

Comment:
yes, this is easier than SQL queries so I now understand you prefer python for the mapping.
Many of the rest of the script has a generic nature we might then to move into a little toolkit in some future